0000-miner-update.md

```diff
@@ -0,0 +1,199 @@
+
+- Why should we *not* do this?
+
+As with most other blockchain-based cryptocurrencies many alternative miner
+implementations are likely to be fueled by a desire to hack or game the system
+for more HNT. While the design of the Helium blockchain should be robust enough
+to resist the most common attacks there will undoubtedly be some bugs or holes
+that were missed. The Helium Blockchain Engineering team will need to respond to
+these exploits in a swift and timely manner so that such an event does not wipe
+out the entire Helium Network.
+
+Diverging implementations of miner are likely to behave slightly differently.
+The result is likely to be unintended forking of the blockchain. Forking is not
+much of a problem for us because it is hard for a fork to make any significant
+progress without consensus from the larger share of the Helium Network. Although
+they may solicit help from the Helium Blockchain Engineering team ultimately it
+will be the responsibility of individual miner package maintainers to resolve
+the blockchain forks and other user complaints resulting from their published
+miner updates.
+
+# Rationale and Alternatives
+[alternatives]: #rationale-and-alternatives
+
+- Why is this design the best in the space of possible designs?
+
+The two miner update designs being considered can be retrofitted on top of our
+existing consumer Helium Hotspots.
+
+- What other designs have been considered and what is the rationale for not
+choosing them?
+
+While there are several third party firmware update services like NervesHub,
+Mender and balena available none of them meet our immediate needs. NervesHub
+relies on Buildroot-based-images and Mender also has support for Buildroot but
+both of these services only allow for monolithic image updates like our existing
+nexthaul/nextgate mechanism. Balena deploys updates via Docker containers but it
+looks like the target edge devices need to run their Yocto-based balenaOS to
+receive them.
+
+- What is the impact of not doing this?
+
+If we do not allow Hotspot owners to build and distribute their own custom miner
+software we eventually risk greater scrutiny by Hotspot operators, regulators
+and large cryptocurrency exchanges.
+
+# Unresolved Questions
+[unresolved]: #unresolved-questions
+
+- What parts of the design do you expect to resolve through the HIP process
+before this gets merged?
+
+What is the minimum viable product we can implement without sacrificing security
+or diminishing the user experience?
+
+- What parts of the design do you expect to resolve through the implementation
+of this feature?
+
+With the exception of image overlays all of the approaches being considered are
+new so the only way to assess them is through rapid prototyping.
+
+- What related issues do you consider out of scope for this HIP that could be
+addressed in the future independently of the solution that comes out of this HIP?
+
+How does a third party package and sign a miner release for download and
```

Comment:
perhaps once you go into third-party mode, no more signing?

Reply:
We could eliminate signature verification.  That increases the risk of a Hotspot being hijacked for a botnet though.